### PR TITLE
#156615347 Update room resource

### PR DESCRIPTION
**Pull request by owenbob**
#### What does this PR do?
- This PR enables a user to update resources in a room
#### How should this be manually tested?
- Run the server
- Test the queries at localhost:5000/mrm
#### What are the relevant pivotal tracker stories?
- #156615347
#### Screenshots (if appropriate)
- Let us see what we have in the database

image

- Let us update the resource with `resourceId`:`1`.

image

- Let us see the result after updating this resource

image

### CODE REVIEW

File: alembic/env.py [hunk 1]
```diff
@@ -15,11 +15,17 @@
...
+import os
+import sys
+sys.path.append(os.getcwd())
+=======
 # target_metadata = None
 import os
 import sys
 sys.path.append(os.getcwd())
 
+>>>>>>> 886bd3f4174947f2fb32aba10b8f635d69cbd92a
```
Comment (proxiex) [inline alembic/env.py]: seems we have a merge conflict here 🤓

File: api/equipment/schema.py [hunk 2]
```diff
@@ -3,12 +3,42 @@
...
\ No newline at end of file
+        interfaces = (relay.Node, )
+
+class UpdateRoomResource(graphene.Mutation):
+    class Arguments:
+        name = graphene.String()
+        room_id = graphene.Int()
+        resource_id = graphene.Int()
+
+    equipment = graphene.Field(Equipment)
```
Comment (proxiex) [inline api/equipment/schema.py]: @owenbob, it looks like you still maintained the use of "equipment". You may need to change it to "resource".

File: tests/test_rooms/test_query_rooms.py [hunk 3]
```diff
@@ -0,0 +1,16 @@
+from tests.base import BaseTestCase
+
+from fixtures.room.room_fixtures import (
+    rooms_query,
+    query_rooms_response,
+    room_query_by_id,
+    room_query_by_id_response
+)
+class QueryRooms(BaseTestCase):
+    def test_query_rooms(self):
```
Comment (shemogumbe) [inline tests/test_rooms/test_query_rooms.py]: You might need to add more test cases to raise test coverage to atleast 70 %

File: api/equipment/schema.py [hunk 4]
```diff
@@ -3,12 +3,42 @@
...
+    def mutate(self,info,room_id,resource_id,**kwargs):
```
Comment (shemogumbe) [inline api/equipment/schema.py]: Would it be ideal to just identify a resource using the resource id and not a combination of resource ID and room ID, remember in updating room resource, we might do things like transfer a resource from one room to another

File: api/equipment/schema.py [hunk 5]
```diff
@@ -3,12 +3,42 @@
...
+        validate_empty_fields(**kwargs)
+
+        query = Equipment.get_query(info)
```
Comment (shemogumbe) [inline api/equipment/schema.py]: Drop the equipment nomenclature in favor of Resource

File: alembic/env.py (hunk 1 above)
Comment (owenbob) [reply alembic/env.py]: Thanks. This has been addressed.

File: api/equipment/schema.py (hunk 2 above)
Comment (owenbob) [reply api/equipment/schema.py]: Thanks for noting this. I have updated it to resources

File: fixtures/location/all_locations_fixtures.py [hunk 6]
```diff
@@ -0,0 +1,81 @@
+
+all_locations_query = '''
+                                    {
+                            allLocations{
+                                edges{
```
Comment (shemogumbe) [inline fixtures/location/all_locations_fixtures.py]: Update this to reflect removed edges and nodes

File: fixtures/location/all_locations_fixtures.py [hunk 7]
```diff
@@ -0,0 +1,81 @@
...
+                                }
+                            }
+                            }
+                                    '''
+
+
+expected_query_all_locations = {
+                                "data": {
+                                    "allLocations": {
+                                    "edges": [
```
Comment (shemogumbe) [inline fixtures/location/all_locations_fixtures.py]: same as above

File: fixtures/room/room_fixtures.py [hunk 8]
```diff
@@ -23,3 +23,56 @@
...
+                {
+                getRoomById(id:1){
+                    capacity
+                    name
+                    roomType
+                }
+                }
+                '''
+
+room_query_by_id_response = {
```
Comment (shemogumbe) [inline fixtures/room/room_fixtures.py]: This response is only valid if the fields in the payload are exactly as this, remember user can request for any set of fields in the payload, you might need more fixtures to reflect the fields users might want in the response data

File: tests/test_resource/test_update_room_equipment.py [hunk 9]
```diff
@@ -0,0 +1,39 @@
+import os
```
Comment (shemogumbe) [inline tests/test_resource/test_update_room_equipment.py]: Let the naming of the file reflect resources and not equipment

File: fixtures/resource/update_resource_fixtures.py [hunk 10]
```diff
@@ -0,0 +1,76 @@
+
+null = None
+
+update_room_resource_query = '''
+                        mutation{
+                        updateRoomResource(roomId:1,resourceId:1,name:"TV Screen"){
```
Comment (shemogumbe) [inline fixtures/resource/update_resource_fixtures.py]: Adjust this to reflect that there is no need to query with roomID, resource ids are enough, room id should just be a field in the kwargs

File: fixtures/resource/update_resource_fixtures.py [hunk 11]
```diff
@@ -0,0 +1,76 @@
...
+                            resource{
+                            name
+                            }
+                        }
+                        }
+                        '''
+
+expected_update_room_resource_query = {
```
Comment (shemogumbe) [inline fixtures/resource/update_resource_fixtures.py]: Do you know we can add success messages to the schema and return it

File: fixtures/location/all_locations_fixtures.py (hunk 6 above)
Comment (owenbob) [reply fixtures/location/all_locations_fixtures.py]: Thanks this has been resolved.

File: fixtures/location/all_locations_fixtures.py (hunk 7 above)
Comment (owenbob) [reply fixtures/location/all_locations_fixtures.py]: Thanks this has been resolved.

File: tests/test_resource/test_update_room_equipment.py (hunk 9 above)
Comment (owenbob) [reply tests/test_resource/test_update_room_equipment.py]: thanks this has been resolved.

File: fixtures/room/room_fixtures.py (hunk 8 above)
Comment (owenbob) [reply fixtures/room/room_fixtures.py]: This has been resolved

File: tests/test_rooms/test_query_rooms.py (hunk 3 above)
Comment (owenbob) [reply tests/test_rooms/test_query_rooms.py]: Thanks have noted this and done accordingly.

File: api/room_resource/schema.py [hunk 12]
```diff
@@ -0,0 +1,46 @@
...
+        return query.all()
+
+class UpdateRoomResource(graphene.Mutation):
+    class Arguments:
+        name = graphene.String()
+        room_id = graphene.Int()
+        resource_id = graphene.Int()
+
+    resource = graphene.Field(Resource)
+    def mutate(self, info, resource_id, **kwargs):
```
Comment (shemogumbe) [inline api/room_resource/schema.py]: LGTM

File: fixtures/room/room_update_fixtures.py [hunk 13]
```diff
@@ -0,0 +1,107 @@
+
+null = None
+
+query_update_all_fields = '''mutation{
+                updateRoom(name:"Entebbe",newName:"Jinja",capacity:8,roomType:"board room"){
```
Comment (shemogumbe) [inline fixtures/room/room_update_fixtures.py]: I really do not like this name=> new name logic, just user kwargs

File: api/room_resource/schema.py [hunk 14]
```diff
@@ -26,6 +30,26 @@ def mutate(self, info, **kwargs):
...
+        resource_id = graphene.Int()
+    resource = graphene.Field(Resource)
+
+    def mutate(self, info, resource_id, **kwargs):
+        validate_empty_fields(**kwargs)
+        query = Resource.get_query(info)
+        exact_resource = query.filter(ResourceModel.id == resource_id).first()
+
+        if not exact_resource:
+            raise GraphQLError("ResourceId not Found")
```
Comment (proxiex) [inline api/room_resource/schema.py]: What do you think about changing *ResourceId not Found* to *Resource not Found*? 🤔

**Comment by proxiex** [comment]
LGTM

File: api/room_resource/schema.py (hunk 14 above)
Comment (owenbob) [reply api/room_resource/schema.py]: Thanks for the feedback. I think they are both appropriate. I just wanted to notify the `front-end dev` what exactly is wrong with their mutation.

Comment (proxiex) [reply api/room_resource/schema.py]: Ok, I get now!

**Pull request merged**